sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2009,12 +2020,200 @@ __ESIMD_API void media_block_store(AccessorTy acc, unsigned x, unsigned y,
 }
 #endif // !__ESIMD_FORCE_STATELESS_MEM
 
+/// Variant of block_load that uses local accessor as a parameter
+/// Loads a contiguous block of memory from given accessor and offset and
+/// returns the loaded data as a vector. Actual code generated depends on
+/// the alignment parameter.
+/// @tparam Tx Element type.
+/// @tparam N Number of elements to load, <code>N * sizeof(Tx)</code> must
+/// be
+///    1, 2, 4 or 8 owords long.
+/// @tparam AccessorTy Accessor type (auto-deduced).
+/// @tparam Flags The alignment specifier type tag. Auto-deduced from the
+///    \c Flags parameter. If it is less than \c 16, then slower unaligned
+///    access is generated, othewise the access is aligned.
+/// @param acc The accessor.
+/// @param offset The offset to load from in bytes.
+/// @return A vector of loaded elements.
+///
+template <typename Tx, int N, typename AccessorTy,
+          typename = std::enable_if_t<
+              !std::is_pointer<AccessorTy>::value &&
+              sycl::detail::acc_properties::is_local_accessor_v<AccessorTy>>>
```

Comment:
Kind of a nitpick but is it possible for both `is_pointer<AccessorTy>` and `is_local_accessor_v<AccessorTy` to be true? It would be a little cleaner if we could just do `is_local_accessor_v<AccessorTy>` and drop the `is_pointer` part.